diff --git a/contrib/check-config.sh b/contrib/check-config.sh
index ac5df62..4eb3e1e 100755
--- a/contrib/check-config.sh
+++ b/contrib/check-config.sh
@@ -174,7 +174,7 @@ echo '- Storage Drivers:'
 	check_flags BLK_DEV_DM DM_THIN_PROVISIONING EXT4_FS EXT4_FS_POSIX_ACL EXT4_FS_SECURITY | sed 's/^/  /'
 
 	echo '- "'$(wrap_color 'overlay' blue)'":'
-	check_flags OVERLAY_FS EXT4_FS_SECURITY EXT4_FS_POSIX_ACL | sed 's/^/  /'
+	check_flags OVERLAYFS_FS EXT4_FS_SECURITY EXT4_FS_POSIX_ACL | sed 's/^/  /'
 } | sed 's/^/  /'
 echo
 
diff --git a/daemon/graphdriver/overlay/overlay.go b/daemon/graphdriver/overlay/overlay.go
index afe12c5..c600a01 100644
--- a/daemon/graphdriver/overlay/overlay.go
+++ b/daemon/graphdriver/overlay/overlay.go
@@ -149,7 +149,7 @@ func supportsOverlay() error {
 
 	s := bufio.NewScanner(f)
 	for s.Scan() {
-		if s.Text() == "nodev\toverlay" {
+		if s.Text() == "nodev\toverlayfs" {
 			return nil
 		}
 	}
@@ -296,11 +296,10 @@ func (d *Driver) Get(id string, mountLabel string) (string, error) {
 	}
 	lowerDir := path.Join(d.dir(string(lowerId)), "root")
 	upperDir := path.Join(dir, "upper")
-	workDir := path.Join(dir, "work")
 	mergedDir := path.Join(dir, "merged")
 
-	opts := fmt.Sprintf("lowerdir=%s,upperdir=%s,workdir=%s", lowerDir, upperDir, workDir)
-	if err := syscall.Mount("overlay", mergedDir, "overlay", 0, label.FormatMountLabel(opts, mountLabel)); err != nil {
+	opts := fmt.Sprintf("lowerdir=%s,upperdir=%s", lowerDir, upperDir)
+	if err := syscall.Mount("overlayfs", mergedDir, "overlayfs", 0, label.FormatMountLabel(opts, mountLabel)); err != nil {
 		return "", fmt.Errorf("error creating overlay mount to %s: %v", mergedDir, err)
 	}
 	mount.path = mergedDir
